.github/workflows/protected-files-validation.yml

```diff
@@ -0,0 +1,81 @@
+name: 'Check Protected Files for Changes'
+
+on:
+  pull_request:
+    paths:
+      - eng/common/**/*
+      - eng/Version.Details.xml
+      - eng/Versions.props
+
+permissions:
+  pull-requests: read
+
+jobs:
+  check-version-details:
+    runs-on: ubuntu-22.04
+    if: ${{ !startsWith(github.event.pull_request.title, 'rebootstrap') && !startsWith(github.event.pull_request.title, 're-bootstrap') && !startsWith(github.event.pull_request.title, 'release') && !startsWith(github.event.pull_request.title, '.NET Source-Build ') }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+
+      - name: Fetch base branch from upstream
+        run: |
+          git fetch origin ${{ github.event.pull_request.base.ref }}
+
+      - name: Check for Version.Details.xml changes
```

Comment:
Is it necessary to explicitly check whether a file change occurred? Given that the workflow only runs on the path filters you've indicated (and it meets the PR title requirements), then you know it's in violation and can simply produce an error without having to check anything more.